Framework/Kernel/inc/MantidKernel/DataService.h

```diff
@@ -19,6 +19,15 @@
 
 namespace Mantid {
 namespace Kernel {
+
+/// Flag for whether to sort items before returning
+enum class DataServiceSort { Sorted, Unsorted };
+/**
+ * Flag for whether to include hidden items when returning,
+ * Auto queries the class to determine this behavior
+ */
+enum class DataServiceHidden { Auto, Include, Exclude };
```

Comment:
When you have an enum that is specific to a class, an alternative is to declare that enum inside the class. That's a way of limiting its scope, similarly as you would do with namespaces. So you would have something like `Mantid::Kernel::DataService::SortOption` or `Mantid::Kernel::DataSerive::HiddenMode`.
I don't necessarily have a strong preference, but maybe this is a very clear-cut case? Notice the names of the enums are `DataService...`.